### ISSUE TITLE
lm() needs do() in modelling.Rmd

### ISSUE THREAD

**Issue by bhive01** [hadley/ggplot2-book#103]
I think I've installed everything and was trying to build the new book (for the first time mind you) and keep hitting a point in modelling.Rmd that I can't get past. 

```shell
Brandons-iMac:~ brandonhurr$ cd ggplot2-book/
Brandons-iMac:ggplot2-book brandonhurr$ make clean
rm -rf book/tex
rm -rf _figures
rm -rf _cache
rm -rf *.html
rm -rf *.pdf
rm -rf book/ggplot2-book.pdf
Brandons-iMac:ggplot2-book brandonhurr$ make -j 2
[snip]
Rscript book/render-tex.R modelling.rmd
Quitting from lines 185-190 (modelling.rmd) 
Error: object 'sales' not found
Calls: <Anonymous> ... mutate -> mutate_ -> mutate_.tbl_df -> mutate_impl -> .Call
Execution halted
make: *** [book/tex/modelling.tex] Error 1
make: *** Waiting for unfinished jobs....
```

Which traces back to this code. 
```r
require(ggplot2) #added so runnable from R directly
# txhousing <- 
txhousing %>% 
  group_by(city) %>% 
  mutate(rel_sales = resid(lm(log(sales) ~ factor(month), 
    na.action = na.exclude))
  )
 # Error: object 'sales' not found
# i tried referencing the vectors directly
txhousing %>% 
  group_by(city) %>% 
  mutate(rel_sales = resid(lm(log(.$sales) ~ factor(.$month), 
    na.action = na.exclude))
  )
# Error: incompatible size (8602), expecting 187 (the group size) or 1
# i tried referencing the data.frame in lm()
txhousing %>% 
  group_by(city) %>% 
  mutate(rel_sales = resid(lm(log(sales) ~ factor(month), 
    data = ., na.action = na.exclude))
  )
# same error as above
# Error: incompatible size (8602), expecting 187 (the group size) or 1
```

All of my searching suggests that you need do() in order to make the lm()'s, but most of the results are from dplyr 0.2. 

```r
> devtools::session_info()
Session info -------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.2.2 (2015-08-14)
 system   x86_64, darwin13.4.0        
 ui       AQUA                        
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       America/Los_Angeles         
 date     2015-11-20                  

Packages -----------------------------------------------------------------------------------------------------
 package    * version     date       source                          
 assertthat   0.1         2013-12-06 CRAN (R 3.2.0)                  
 colorspace   1.2-6       2015-03-11 CRAN (R 3.2.0)                  
 DBI          0.3.1       2014-09-24 CRAN (R 3.2.0)                  
 devtools     1.9.1.9000  2015-11-16 Github (hadley/devtools@b4edf3e)
 digest       0.6.8       2014-12-31 CRAN (R 3.2.0)                  
 dplyr      * 0.4.3.9000  2015-11-16 Github (hadley/dplyr@4f2d7f8)   
 ggplot2    * 1.0.1.9003  2015-11-19 Github (hadley/ggplot2@23a748f) 
 gtable       0.1.2       2012-12-05 CRAN (R 3.2.0)                  
 labeling     0.3         2014-08-23 CRAN (R 3.2.0)                  
 lazyeval     0.1.10.9000 2015-05-26 Github (hadley/lazyeval@ecb8dc0)
 magrittr     1.5         2014-11-22 CRAN (R 3.2.0)                  
 memoise      0.2.1       2014-04-22 CRAN (R 3.2.0)                  
 munsell      0.4.2       2013-07-11 CRAN (R 3.2.0)                  
 plyr         1.8.3.9000  2015-10-16 Github (hadley/plyr@9d8294e)    
 R6           2.1.1       2015-08-19 CRAN (R 3.2.0)                  
 Rcpp         0.12.2      2015-11-15 CRAN (R 3.2.2)                  
 readxl     * 0.1.0.9000  2015-11-02 Github (hadley/readxl@7c7f66b)  
 reshape2     1.4.1       2014-12-06 CRAN (R 3.2.0)                  
 scales       0.3.0.9000  2015-11-19 Github (hadley/scales@7fc502c)  
 stringi      1.0-1       2015-10-22 CRAN (R 3.2.0)                  
 stringr      1.0.0       2015-04-30 CRAN (R 3.2.0)                  
 tidyr      * 0.3.1.9000  2015-11-19 Github (hadley/tidyr@c714c72)   
```

**Comment by hadley** [comment]
This seems to be a scoping bug in dev dplyr. Works with released version.

cc @romainfrancois

**Comment by bhive01** [comment]
Tried to build the book again and ran into these issues, which were fixed with the code changes suggested here: #106 and #104 
Loaded up CRAN binary for dplyr and got stuck on this one again. 
```r
Rscript book/render-tex.R modelling.rmd
Quitting from lines 185-190 (modelling.rmd) 
Error in eval(expr, envir, enclos) : 
  incompatible size (8034), expecting 187 (the group size) or 1
Calls: <Anonymous> ... mutate -> mutate_ -> mutate_.tbl_df -> mutate_impl -> .Call
Execution halted
make: *** [book/tex/modelling.tex] Error 1

Exited with status 2.
```
